src/test/java/co/rsk/federate/signing/hsm/advanceblockchain/ConfirmedBlocksProviderTest.java

```diff
@@ -181,152 +182,24 @@ void getConfirmedBlocksHSMVersion4BelowDifficultyCap() {
-        Block mockBestBlock = TestUtils.mockBlock(18, TestUtils.createHash(18));
-        when(mockBlockStore.getBestBlock()).thenReturn(mockBestBlock);
-
-        List<BlockHeader> brothers = Arrays.asList(
-            blockHeaderBuilder.setNumber(1).setDifficulty(new BlockDifficulty(BigInteger.valueOf(5))).build(),
-            blockHeaderBuilder.setNumber(2).setDifficulty(new BlockDifficulty(BigInteger.valueOf(10))).build()
-        );
-
-        for (int i = 11; i < 19; i++) {
-            long difficultyValue = 15;
-            Block mockBlockToProcess = TestUtils.mockBlockWithUncles(i, TestUtils.createHash(i), difficultyValue, brothers);
-            when(mockBlockStore.getChainBlockByNumber(i)).thenReturn(mockBlockToProcess);
-        }
-
-        // Below Difficulty Cap For HSM 2
-        ConfirmedBlocksProvider confirmedBlocksProvider = new ConfirmedBlocksProvider(
-            new BigInteger("160"),
-            100,
-            mockBlockStore,
-            difficultyCapRegTest,
-            HSMVersion.V2
-        );
-
-        List<Block> confirmedBlocks = confirmedBlocksProvider.getConfirmedBlocks(startingPoint);
-
-        // HSM 2 Doesn't consider brothers difficulty
-        // Assert 0 element in confirmed list
-        assertEquals(0, confirmedBlocks.size());
-
-        // Above Difficulty Cap For HSM 3
-        confirmedBlocksProvider = new ConfirmedBlocksProvider(
-            new BigInteger("160"),
-            100,
-            mockBlockStore,
-            difficultyCapRegTest,
-            HSMVersion.V4
-        );
-
-        confirmedBlocks = confirmedBlocksProvider.getConfirmedBlocks(startingPoint);
-
-        // HSM 3 considers brothers difficulty
-        // Assert 1 element in confirmed and 7 in potential list
-        assertEquals(8, confirmedBlocks.size());
-    }
-
-    @Test
-    void getBlockDifficultyToConsider_forHSMVersionLessThan4_doesNotConsiderUnclesAndCapDifficulty() {
-        // arrange
-        Block block = buildBlockWithUncles();
-
-        // build blocks provider for hsm version 2
-        ConfirmedBlocksProvider confirmedBlocksProvider = new ConfirmedBlocksProvider(
-            BigInteger.valueOf(160),
-            100,
-            mock(BlockStore.class),
-            MAINNET.getDifficultyCap(),
-            HSMVersion.V2
-        );
-
-        // act
-        BigInteger consideredDifficulty = confirmedBlocksProvider.getBlockDifficultyToConsider(block);
-
-        // assert
-        // HSM version less than 4 does not consider brothers nor cap difficulty
-        // = 7000000000000000000001 difficulty from block 4
-        BigInteger expectedConsideredDifficulty = new BigInteger("7000000000000000000001");
-        assertEquals(expectedConsideredDifficulty, consideredDifficulty);
-    }
-
-    @Test
-    void getBlockDifficultyToConsider_forHSMVersionMoreThan4_considersUnclesAndCapDifficulty() {
+    void getBlockDifficultyToConsider_forHSMVersionMoreThan5_considersUnclesAndCapDifficulty() {
```

Comment:
```suggestion
    void getBlockDifficultyToConsider_considersUnclesAndCapDifficulty() {
```
no big deal, but i think you removed all _versions_ references in tests naming